src/process/process.rs

```diff
@@ -164,24 +161,22 @@ impl<'a> Process<'a> {
             // Set kernel break to beginning of callback buffer
             kernel_memory_break =
                 kernel_memory_break.offset(-(callback_offset as isize));
-            let callback_buf = mem::transmute(Slice {
-                data: kernel_memory_break as *const Option<Callback>,
-                len: callback_len
-            });
+            let callback_buf = slice::from_raw_parts_mut(
+                kernel_memory_break as *mut Callback, callback_len);
 
             let callbacks = RingBuffer::new(callback_buf);
 
-            let load_result = load(start_addr, memory.data);
+            let load_result = load(start_addr, memory.as_mut_ptr());
 
             let stack_bottom = load_result.app_mem_start.offset(512);
 
             let mut process = Process {
                 memory: memory,
                 app_memory_break: stack_bottom,
                 kernel_memory_break: kernel_memory_break,
-                text: Slice {
-                    data: start_addr.offset(-1) as *const u8,
-                    len: length as usize },
+                text: slice::from_raw_parts(
+                    start_addr.offset(-1) as *const u8,
```

Comment:
Since `start_addr` is now a `*const u8` instead of `*const usize` I believe this should be `-4` rather than `-1`. I think this isn't manifesting as a bug because we currently only use this for the MPU config anyway, which masks off the lower bits of the base address. Thoughts @yuriks? I could be missing something...